app/controllers/provider_interface/provider_interface_controller.rb

```diff
@@ -14,6 +14,13 @@ class ProviderInterfaceController < ActionController::Base
       render template: 'provider_interface/account_creation_in_progress', status: :forbidden
     }
 
+    rescue_from MissingPermission, with: lambda { |e|
```

Comment:
Should we have a test for this behaviour? 

I think you could rescue with a method called `missing_permission` here and it would know which view to use? Would be good not to need a separate controller.

Reply:
I've added a 'respond to an application' scenario covering the case of hand-crafting urls exercising this path. I've also removed the separate directory and started rescuing with a named method, but specifying the template is still needed.